internal/instrument.go

```diff
@@ -0,0 +1,145 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal
+
+import (
+	"fmt"
+	"go/parser"
+	"go/token"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/dave/dst"
+	"github.com/dave/dst/decorator"
```

Comment:
I think we need ut for this

Reply:
@123liuziming  Do you mean unit test?

Reply:
This is a prototype implementation. Let's focus on the most critical injection mechanism first, and then add unit tests.